### DISCUSSION TITLE
post/ai_written_programs/

### DISCUSSION THREAD

**Discussion by swilcox** [swilcox/swilcox.github.io#3] [General]
https://swilcox.github.io/post/ai_written_programs/ 

 What does it look like when you attempt to write a whole program with AI?

**Comment by bcorfman** [comment]
Thanks for the mention. I had considered trying the same task with ChatGPT, but my article was already long enough! Now that I see the ChatGPT code, I can definitively say: Yuck. :smile: I don't like its coding style, plus it's hard to grok. The verbosity doesn't help either. BTW, did you use 4o or an earlier version? I noticed a large positive jump in quality with 4o when I tried it on other coding problems, so I'd be surprised if it generated that same result.

**Reply by swilcox** [reply]
It was the desktop version of ChatGPT, but free edition, so `ChatGPT-4-Turbo`.

**Comment by bcorfman** [comment]
Also, your comment is spot on about both LLMs not being up to this task.  I was still impressed that it saved me a load of time though. I wouldn't have been able to achieve such great initial quality out of the gate. I guess that's why I was more forgiving in my assessment. I agree though -- we developers better maintain our knowledge and reasoning abilities to be able to evaluate what the AIs are doing. For now, anyway.
